tests/utils_tests/test_timezone.py

```diff
@@ -54,6 +57,24 @@ def test_now(self):
         with override_settings(USE_TZ=False):
             self.assertTrue(timezone.is_naive(timezone.now()))
 
+    def test_localdate(self):
+        naive = datetime.datetime(2015, 1, 1, 0, 0, 1)
+        with self.assertRaisesMessage(ValueError, 'astimezone() cannot be applied to a naive datetime'):
```

Comment:
See also my recent Trac ticket suggesting to normalize behavior across Python versions.